src/dotty/runtime/vc/VCBooleanPrototype.scala

```diff
@@ -0,0 +1,58 @@
+package dotty.runtime.vc
+
+import scala.reflect.ClassTag
+
+import scala.runtime.Statics
+
+abstract class VCBooleanPrototype(val underlying: Boolean) extends VCPrototype {}
+
+abstract class VCBooleanCasePrototype(underlying: Boolean) extends VCBooleanPrototype(underlying) with Product1[Boolean] {
+
+  final def _1: Boolean = underlying
+
+  override final def hashCode(): Int = {
+    underlying.hashCode()
+  }
+
+  override final def toString: String = {
+    s"$productPrefix($underlying)"
+  }
+
+  // subclasses are expected to implement equals, productPrefix, and canEqual
+}
+
+abstract class VCBooleanCompanion[T <: VCBooleanPrototype] extends ClassTag[T] {
+  def box(underlying: Boolean): T
+  final def unbox(boxed: T) = boxed.underlying
+  override def newArray(len: Int): Array[T] =
+    new VCBooleanArray(this, len).asInstanceOf[Array[T]]
+
+
+  final def _1$extension(underlying: Boolean)       = underlying
+  final def hashCode$extension(underlying: Boolean) = underlying.hashCode()
+  final def toString$extension(underlying: Boolean) = s"${productPrefix$extension(underlying)}($underlying)"
+  def productPrefix$extension(underlying: Boolean): String
```

Comment:
Value classes which are not case classes will not implement `productPrefix`, this should not be here.